src/iter/internal.rs

```diff
@@ -166,6 +172,34 @@ impl Splitter {
     }
 }
 
+#[derive(Clone, Copy)]
+struct LengthSplitter {
+    inner: Splitter,
+    min: usize,
+}
+
+impl LengthSplitter {
+    #[inline]
+    fn new(min: usize, max: usize, len: usize) -> LengthSplitter {
+        let mut splitter = LengthSplitter {
+            inner: Splitter::new(),
+            min: cmp::max(min, 1),
+        };
+
+        let max_splits = len / cmp::max(max, 1);
```

Comment:
I'm a bit confused by the `max_splits` computation here, actually. So in particular if the default is that `max` is `usize::MAX`, then wouldn't `max_splits` almost always be `0`? I guess tomorrow I'll have to trace this out in more detail.